### PR TITLE
Send workspace folder diff upon listener notification

### PR DESCRIPTION
**Pull request by tsmaeder**
This is a fix for 

Workspace folder changes are not sent to the server if they occur during initialisation #451

The gist of the fix is to remember the workspace folders sent to a language server during the "initialize" call and then send the diff between that state and the current WS folders when registering the workspace change listener.

Signed-off-by: Thomas Mäder <tmader@redhat.com>

### CODE REVIEW

**Comment by tsmaeder** [comment]
@dbaeumer I cannot seem to sign the CLA: I'm always stuck on the "sign into github to sign" page or if I'm forwarded I end up at "github.com/" with seemingly no action being taken. Tried with FF and Chromium both times in incognito mode. Who can I complain to?

**Comment by tsmaeder** [comment]
Fixed lint errors

File: client/src/workspaceFolders.ts [hunk 1]
```diff
@@ -82,21 +92,40 @@ export class WorkspaceFoldersFeature implements DynamicFeature<undefined> {
 				id: id,
 				registerOptions: undefined
 			});
+			this.sendInitialEvent(workspace.workspaceFolders);
```
Comment (dbaeumer) [inline client/src/workspaceFolders.ts]: To make this consistent with other feature (e.g. DidOpenTextDocumentFeature) could we send the extra event in the register hook. Then we always have the same code pathes.

**Comment by dbaeumer** [comment]
@tsmaeder thanks for the PR. I have one minor request. See above.

File: client/src/workspaceFolders.ts (hunk 1 above)
Comment (tsmaeder) [reply client/src/workspaceFolders.ts]: like so?

**Pull request merged**